backend/tests/unit/core/ipam/conftest.py

```diff
@@ -1,17 +1,88 @@
+from pathlib import Path
+
 import pytest
 
+from infrahub import config
 from infrahub.core import registry
 from infrahub.core.branch import Branch
 from infrahub.core.constants import InfrahubKind
+from infrahub.core.diff.repository.repository import DiffRepository
+from infrahub.core.initialization import (
+    create_default_branch,
+    create_global_branch,
+    create_ipam_namespace,
+    create_root_node,
+)
 from infrahub.core.node import Node
+from infrahub.core.query.delete import DeleteAfterTimeQuery
+from infrahub.core.schema import SchemaRoot, core_models, internal_schema
+from infrahub.core.schema.manager import SchemaManager
 from infrahub.core.schema.schema_branch import SchemaBranch
+from infrahub.core.timestamp import Timestamp
+from infrahub.core.utils import delete_all_nodes
 from infrahub.database import InfrahubDatabase
+from infrahub.dependencies.registry import get_component_registry
 
 
-@pytest.fixture
-async def ip_dataset_01(
+@pytest.fixture(scope="module")
+async def register_internal_models_schema(default_branch: Branch) -> SchemaBranch:
+    schema = SchemaRoot(**internal_schema)
+    schema_branch = registry.schema.register_schema(schema=schema, branch=default_branch.name)
+    default_branch.update_schema_hash()
+    return schema_branch
+
+
+@pytest.fixture(scope="module")
+async def register_core_models_schema(default_branch: Branch, register_internal_models_schema) -> SchemaBranch:
+    schema = SchemaRoot(**core_models)
+    schema_branch = registry.schema.register_schema(schema=schema, branch=default_branch.name)
+    default_branch.update_schema_hash()
+    return schema_branch
+
+
+@pytest.fixture(scope="module")
+def local_storage_dir(tmp_path_module_scope: Path) -> Path:
+    storage_dir = tmp_path_module_scope / "storage"
+    storage_dir.mkdir()
+
+    config.SETTINGS.storage.driver = config.StorageDriver.FileSystemStorage
+    config.SETTINGS.storage.local.path_ = storage_dir
+
+    return storage_dir
+
+
+@pytest.fixture(scope="module")
+async def default_branch(local_storage_dir: str, db: InfrahubDatabase) -> Branch:
```

Comment:
Should function body be factorized into a new function to avoid duplication of `backend.tests.helpers.default_branch` ?

Reply:
good suggestion, I've made that change